tasks/process_flakes.py

```diff
@@ -53,18 +54,30 @@ def run_impl(
         """
         log.info(
             "Received process flakes task",
-            extra=dict(repoid=repo_id, commit=commit_id),
+            extra=dict(repoid=repo_id),
         )
 
-        redis_client = get_redis_connection()
         lock_name = f"flake_lock:{repo_id}"
+        list_key = f"flake_uploads:{repo_id}"
+        process_func = process_flake_for_repo_commit
+
         try:
+            redis_client = get_redis_connection()
             with redis_client.lock(
                 lock_name, timeout=max(300, self.hard_time_limit_task), blocking=False
             ):
-                while redis_client.get(f"flake_uploads:{repo_id}") is not None:
-                    redis_client.delete(f"flake_uploads:{repo_id}")
-                    process_flake_for_repo_commit(repo_id, commit_id)
+                try:
+                    while (commit_ids := redis_client.lpop(list_key, 10)) is not None:
+                        for commitid in commit_ids:
+                            process_func(repo_id, commitid.decode())
+
+                except ResponseError as e:
+                    if "WRONGTYPE" in str(e) and commit_id:
+                        while redis_client.getdel(list_key) is not None:
```

Comment:
as the corresponding `rpush` has a similar compatibility code, this might run into a race condition:

- this task tries to `lpop`, runs into an exception
- the finisher tries to `lpush`, runs into an exception
- the finisher completes its "delete + rpush" first
- we then run the `getdel` here

I believe you can avoid this backwards compatibility code by just using a different redis key (though still make sure that the `flake_uploads` one is being deleted, as I believe it is not defined with a TTL so might hang around for eternity otherwise.